cypress/e2e/shared/notificationRules.test.ts

```diff
@@ -292,4 +302,457 @@ describe('NotificationRules', () => {
+      cy.getByTestID('overlay').should('not.exist')
+
+      cy.getByTestID('check-status-dropdown').should('not.exist')
+      cy.getByTestID('check-status-input').click()
+      cy.getByTestID('check-status-dropdown').should('be.visible')
+
+      // Filter level == crit
+      cy.getByTestID('check-status-input')
+        .clear()
+        .type('"level" == "crit"')
+      cy.get('.event-row').should('have.length', 1)
+      //    ...verify row
+      cy.get('[class$=ScrollContainer] > div:nth-of-type(1)').within(() => {
+        cy.get('.level-table-field--crit').should('be.visible')
+        cy.get('.sent-table-field--sent').should('be.visible')
+      })
+      // Filter level != info
+      cy.getByTestID('check-status-input')
+        .clear()
+        .type('"level" != "info"')
+      cy.get('.event-row')
+        .should('have.length', 3)
+        .then(rows => {
+          const levels = rows
+            .find('.event-row--field:nth-of-type(2)')
+            .toArray()
+            .map(r => r.innerText)
+          expect(levels).to.deep.eq(['ok', 'crit', 'warn'])
+        })
+      // Filter notificationRuleName == rule.name
+      cy.getByTestID('check-status-input')
+        .clear()
+        .type(`"notificationRuleName" == "${rule.name}"`)
+      cy.get('.event-row')
+        .should('have.length', 4)
+        .then(rows => {
+          const levels = rows
+            .find('.event-row--field:nth-of-type(2)')
+            .toArray()
+            .map(r => r.innerText)
+          expect(levels).to.deep.eq(['ok', 'info', 'crit', 'warn'])
+        })
+      cy.getByTitle('Refresh').click()
+      cy.getByTestID('check-status-dropdown').should('not.exist')
+    })
+
+    it('refreshes the history', () => {
+      cy.get('.event-row').should('have.length', 4)
+      cy.get<GenCheck>('@check').then(check => {
+        cy.get<NotificationEndpoint>('@endp').then(endp => {
+          cy.get<GenRule>('@rule').then(rule => {
+            writeMonitoringRecord(
+              {
+                check: check,
+                endp: endp,
+                rule: rule,
+                level: 'info',
+                source: 'wumpus',
+                sent: 'false',
+                valField: {key: 'foo', val: 'bar'},
+                sourceTimestamp: calcNanoTimestamp('4m'),
+                statusTimestamp: calcNanoTimestamp('3m'),
+                message: 'This is a fake info message',
+              },
+              '2m'
+            )
+          })
+        })
+      })
+      cy.getByTitle('Refresh').click()
+      cy.get('.event-row')
+        .should('have.length', 5)
+        .then(rows => {
+          const levels = rows
+            .find('.event-row--field:nth-of-type(2)')
+            .toArray()
+            .map(r => r.innerText)
+          expect(levels).to.deep.eq(['info', 'ok', 'info', 'crit', 'warn'])
+        })
+    })
+
+    // N.B. if local === utc then skip
+    if (new Date().getTimezoneOffset() !== 0) {
```

Comment:
instead of this we could modify `package.json` scripts to set the `TZ` variable like this: https://github.com/cypress-io/cypress/issues/1043#issuecomment-623293020

Reply:
Ah rats ok. We have a new ability in our pipeline to run tests 20x so we could do that to check for flakiness before merging.